### PR TITLE
Added source File name for emitAsset func

### LINKED ISSUE #13695: Improve emitAsset error with sourceFilename
## Feature request

Currently assets with the same filenames provide an error like `Conflict: Multiple assets emit different content to the same filename ${file}`. This error includes final file destination (like `assets/smth.svg` e.x.) but there is no way to know what particular local files lead to this.
https://github.com/webpack/webpack/blob/c181294865dca01b28e6e316636fef5f2aad4eb6/lib/Compilation.js#L3506

**What is the expected behavior?**

Error message should provide info about what file has same final filename.

**What is motivation or use case for adding/changing the behavior?**

It's hard to search through dozen files with same name in monorepo to find that single one that conflicts with other.

**How should this be implemented in your opinion?**

Add `assetInfo.sourceFilename` to the error to help find files.

**Are you willing to work on this yourself?**
nope

### PR DESCRIPTION
**Pull request by g4rry420**
Signed-off-by: Gurkiran Singh <gurkiransinghk@gmail.com>

**What kind of change does this PR introduce?**
Added the source File Info for emitAsset func .
Resolves #13695 
Since, this is my first pr to webpack, I wasn't sure how to test this after making the changes. If someone can please let me know, that be really helpful.

**Did you add tests for your changes?**
No

**Does this PR introduce a breaking change?**
No

**What needs to be documented once your changes are merged?**
Nothing

### CODE REVIEW

File: lib/Compilation.js [hunk 1]
```diff
@@ -4086,7 +4086,7 @@ This prevents using hashes of each other and should be avoided.`);
 			if (!isSourceEqual(this.assets[file], source)) {
 				this.errors.push(
 					new WebpackError(
-						`Conflict: Multiple assets emit different content to the same filename ${file}`
+						`Conflict: Multiple assets emit different content to the same filename ${file}. Source File: ${source}`
```
Comment (sokra) [inline lib/Compilation.js:4089]: `source` is the content of the file.

You could include `assetInfo` in the message instead, this might give hints.

**Review by sokra — CHANGES_REQUESTED** [review]

File: lib/Compilation.js [hunk 2]
```diff
@@ -4086,7 +4086,7 @@ This prevents using hashes of each other and should be avoided.`);
 			if (!isSourceEqual(this.assets[file], source)) {
 				this.errors.push(
 					new WebpackError(
-						`Conflict: Multiple assets emit different content to the same filename ${file}`
+						`Conflict: Multiple assets emit different content to the same filename ${file}. Source File: ${assetInfo}`
```
Comment (sokra) [inline lib/Compilation.js:4089]: You can't print an object that way, you might want to `JSON.stringify` it, or print it in a nicer way.

Anyway, please also make sure to test the code. e. g. here is a test case you can update: `test/configCases/emit-asset/different-source`

**Review by sokra — CHANGES_REQUESTED** [review]

File: lib/Compilation.js (hunk 2 above)
Comment (g4rry420) [reply lib/Compilation.js]: My apologies.
For testing, I have run the `yarn jest ConfigTestCases` and from my understanding the testing has been done in `ConfigTestCases.tempate.js` file which grabs the files from `configCases` folder and tests them. But still didn't understand what should I update in `test/configCases/emit-asset/different-source` ? Please, let me know about it.

**Comment by alexander-akait** [comment]
Fixed by https://github.com/webpack/webpack/pull/15888

**Pull request closed without merging**